[Semaphore] Add documentation about semaphore configuration (#16767)
symfony/symfony-docs: reference/configuration/framework.rst

Name of the lock you want to create.

semaphore
~~~~~~~~~

.. versionadded:: 6.1

    The ``semaphore`` option was introduced in Symfony 6.1.

**type**: ``string`` | ``array``

The default semaphore adapter. Store's DSN are also allowed.

> .. _reference-lock-enabled:

Reviewer: ```suggestion
.. _reference-semaphore-enabled:
```

enabled
.......

**type**: ``boolean`` **default**: ``true``

Whether to enable the support for semaphore or not. This setting is
automatically set to ``true`` when one of the child settings is configured.

> .. _reference-lock-resources:

Reviewer: ```suggestion
.. _reference-semaphore-resources:
```

resources
.........

**type**: ``array``

> A list of semaphore stores to be created by the framework extension.

Reviewer: ```suggestion
A map of semaphore stores to be created by the framework extension, with
the name as key and DSN as value:
```

.. configuration-block::

    .. code-block:: yaml

        # config/packages/semaphore.yaml
        framework:

>             semaphore: '%env(SEMAPHORE_DSN)%'

Reviewer: ```suggestion
            semaphore:
                resources:
                    default: '%env(SEMAPHORE_DSN)%'
```

Author: Both are correct, but the first one (`semaphore: '%env(SEMAPHORE_DSN)%'`) is easiest to read IMHO and consistent with `lock` documentation

    .. code-block:: xml

        <!-- config/packages/lock.xml -->
        <?xml version="1.0" encoding="UTF-8" ?>
        <container xmlns="http://symfony.com/schema/dic/services"
            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
            xmlns:framework="http://symfony.com/schema/dic/symfony"
            xsi:schemaLocation="http://symfony.com/schema/dic/services
                https://symfony.com/schema/dic/services/services-1.0.xsd
                http://symfony.com/schema/dic/symfony https://symfony.com/schema/dic/symfony/symfony-1.0.xsd">

            <framework:config>
                <framework:semaphore>

>                     <framework:resource>%env(SEMAPHORE_DSN)%</framework:resource>

Reviewer: ```suggestion
                    <framework:resource name="default">%env(SEMAPHORE_DSN)%</framework:resource>
```
